apps/examples/board_specific_demo/ameba_secure_storage_demo/secure_storage.c

```diff
@@ -63,88 +66,160 @@ int sstorage_main(int argc, char *argv[])
 	security_handle hnd;
 	security_error res;
 	security_data data;
-	uint32_t slot;
+	uint32_t input_data;
+	uint32_t slot_index;
 	uint32_t address;
 	uint32_t length = TEST_DATA_LENGTH;
 	char ss_path[7];
+	int option;
 
 	printf("Start testing SE Secure Storage!!\n");
+
+	while ((option = getopt(argc, argv, "d:s:h")) > 0) {
+		switch (option) {
+		case 'd':
+			sscanf(optarg, "%x", &input_data);
+			break;
+		case 's':
+			slot_index = atoi(optarg);
+			if (slot_index > 31) {
+				slot_index = 0;
+			}
+			break;
+		case 'h':
+			printf("Usage command: -\n sstorage -s <slot index> -d <hex data value>\n");
+			return 0;
+		default:
+			break;
+		}
+	}
+
+	if (argc < 4) {
+		input_data = TEST_DEFAULT_DATA;
+		slot_index = TEST_SLOT_NUM;
+	}
+
+	printf("Data: %2x, slot: %d\n", input_data, slot_index);
 	printf("Start Security init!!\n");
-	sleep(1);
 	res = security_init(&hnd);
 	if (res != SECURITY_OK) {
 		printf("security_init() Fail\n", __LINE__);
 		sleep(1);
 	}
+	printf("Security init complete\n");
+	sleep(1);
+
+	data.length = TEST_DATA_LENGTH;
+
+	/* Form the SS path name based on the slot index where we want to read and write */
+	snprintf(ss_path, 7, "ss/%d", slot_index);
+
+	printf("Let us read once before writing to the slot\n");
+	printf("ss_read_secure_storage() Slot: %d\n\n", slot_index);
+	data.data = (void *)pre_read;
+	res = ss_read_secure_storage(hnd, ss_path, 0, &data);
+	if (res != SECURITY_OK) {
+		printf("ss_read_secure_storage() Fail\n", __LINE__);
```

Comment:
Need to add `%d` to print the line number. Also you could use `__func__`  to print the function name.

Reply:
Done